js/src/transaction-builder.ts

```diff
@@ -332,6 +333,27 @@ export class TransactionBuilder<ResponseType> {
     return info.partialFee.toBigInt();
   }
 
+  /**
+   * ## Decode response payload from sign and send transaction
+   * @param payload - Raw bytes from the reply message
+   * @returns Decoded payload
+   */
+  public decodePayload(payload: Bytes): ResponseType {
+    const method = getPayloadMethod(this._responseType);
+    const noPrefixPayload = payload.slice(this._prefixByteLength);
+    const type = this._registry.createType<any>(this._responseType, noPrefixPayload);
+
+    return type[method]();
+  }
+
+  /**
+   * ## Check if the reply is an error and throw if it is
+   * @param message - UserMessageSent message
+   */
+  public checkErrorReply({ payload, details }: GearCoreMessageUserUserMessage) {
```

Comment:
Well, probably `throwOnErrorReply` method name would be better, or at least `handleErrorReply`, since `check` suggests boolean return